### DISCUSSION TITLE
What is the name of the UI component that renders the server response and converts the MD

### DISCUSSION THREAD

**Discussion by JJRobertsSex30** [mckaywrigley/chatbot-ui#1767] [Help]
Hi I noticed that when the LLM responds to the client using markdown language it appears as markdown for a split seconds before being correctly formatted.

What is the component / part of the code that is doing this ? Is it an external library / component or a bespoke part of this code ?

**Comment by castortech** [comment]
You can start looking at processMarkdown in lib/retrieval/processing/md.ts
